File: 11. Container With Most Water/solution.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using std::max;
using std::min;
using std::vector;

class Solution {
    public:
        int maxArea(vector<int>& height) {
            int i = 0, j = height.size() - 1, ret = 0;
            int h = 0;
            while (i < j)
            {
                h = min(height[i], height[j]);
                ret = max(ret, h * (j - i));
                while(height[i] <= h && i < j) i++;
                while(height[j] <= h && i < j) j--;
            }

            return ret;
        }
};

int main() {
    return 0;
}

File: 11. Container With Most Water/solution.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """

        l = 0
        r = len(height) - 1
        res = 0

        while(l < r):
            h = min(height[l], height[r])
            res = max(res, (r - l) * h)

            while(height[r] <= h and l < r):
                r -= 1
            while(height[l] <= h and l < r):
                l += 1

        return res
